_posts/en/newsletters/2021-04-21-newsletter.md

```diff
@@ -0,0 +1,218 @@
+    preventing spending the last of your onchain funds (even to open a
+    new channel) while you still have at least one channel open.
+
+    Osuntokun's request is for wallets or services built on LND to let
+    the development team know if any of the above concerns, or any other
+    concerns related to anchor outputs, will cause serious problems.
+    Although the question is specific to LND, the answers may have
+    implications for all LN nodes.
+
+- **Sapio public launch:** Jeremy Rubin [posted][rubin sapio] to the
+  Bitcoin-Dev mailing list an announcement that he has made available
+  the Sapio smart contract development toolkit, a Rust-based library
+  and associated tooling that can be used to create smart contracts
+  expressible using Bitcoin Script.  The language was originally
+  designed to make use of Rubin's proposed
+  [OP_CHECKTEMPLATEVERIFY][topic op_checktemplateverify] opcode
+  (`OP_CTV`) but it can simulate the existence of that opcode, and of
+  other potential features for Bitcoin such as taproot, using a trusted
+  signing oracle.  In addition to the Sapio library, the release also
+  contains extensive documentation and an experimental frontend.
+
+## Changes to services and client software
+
+*In this monthly feature, we highlight interesting updates to Bitcoin
+wallets and services.*
+
+FIXME:bitschmidty
+
+## Releases and release candidates
+
+*New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+- [Bitcoin Core 0.21.1rc1][Bitcoin Core 0.21.1] is a release candidate
+  for a version of Bitcoin Core that, if activated, will enforce the
+  rules of the proposed [taproot][topic taproot] soft fork, which uses
+  [schnorr signatures][topic schnorr signatures] and allows use of
+  [tapscript][topic tapscript].  These are, respectively, specified by
+  BIPs [341][BIP341], [340][BIP340], and [342][BIP342].  Also included
+  is the ability to pay [bech32m][topic bech32] addresses specified by
+  [BIP350][], although bitcoins spent to such addresses on mainnet will
+  not be secure until a soft fork such as taproot is activated.  The
+  release additionally includes bug fixes and minor improvements.
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
+[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1
+repo], [Hardware Wallet Interface (HWI)][hwi repo],
+[Rust Bitcoin][rust bitcoin repo], [BTCPay Server][btcpay server repo],
+[Bitcoin Improvement Proposals (BIPs)][bips repo], and [Lightning
+BOLTs][bolts repo].*
+
+- [Bitcoin Core #21377][] adds the activation mechanism and [#21686][Bitcoin
+  Core #21686] adds the activation parameters for the Taproot softfork.
+Starting with the first difficulty adjustment after April 24th, miners will be
```

Comment:
I just noticed that my editor here wasn't set up to indent the block here. Will fix.